internal/utils/main.go

```diff
@@ -757,28 +795,42 @@ func CreateLedgerBackend(ctx context.Context, useCaptiveCore bool, env Environme
 	}
 
 	// Create ledger backend from datastore
-	fileConfig := ledgerbackend.LCMFileConfig{
-		StorageURL:        env.StorageURL,
-		FileSuffix:        ".xdr.gz",
-		LedgersPerFile:    1,
-		FilesPerPartition: 64000,
+	params := make(map[string]string)
+	params["destination_bucket_path"] = env.DatastorePath
+	dataStoreConfig := datastore.DataStoreConfig{
+		Type:   "GCS",
+		Params: params,
 	}
 
-	parsed, err := url.Parse(env.StorageURL)
+	dataStore, err := datastore.NewDataStore(ctx, dataStoreConfig, env.Network)
 	if err != nil {
 		return nil, err
 	}
 
-	// Using the GCS datastore backend
-	if parsed.Scheme == "gcs" {
-		backend, err := ledgerbackend.NewGCSBackend(ctx, fileConfig)
-		if err != nil {
-			return nil, err
-		}
-		return backend, nil
+	// TODO: In the future these will come from a config file written by ledgerexporter
+	// Hard code ledger batch values for now
+	ledgerBatchConfig := datastore.LedgerBatchConfig{
+		LedgersPerFile:    1,
+		FilesPerPartition: 64000,
+		FileSuffix:        ".xdr.gz",
+	}
+
+	// TODO: In the future CompressionType should be removed as it won't be configurable
+	BSBackendConfig := ledgerbackend.BufferedStorageBackendConfig{
+		LedgerBatchConfig: ledgerBatchConfig,
+		CompressionType:   "gzip",
+		DataStore:         dataStore,
+		BufferSize:        1000,
+		NumWorkers:        5,
+		RetryLimit:        3,
+		RetryWait:         5,
```

Comment:
If these values are hardcoded here, why are we passing in as flags?

Reply:
That's because I forgot to update the code to use the new flags 😆